app/src/main/java/ru/suslovalex/androidacademyapp/data/JsonLoad.kt

```diff
@@ -0,0 +1,109 @@
+package ru.suslovalex.androidacademyapp.data
+
+import android.content.Context
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+
+
+private val jsonFormat = Json { ignoreUnknownKeys = true }
+
+@Serializable
+private class JsonGenre(val id: Int, val name: String)
+
+@Serializable
+private class JsonActor(
+    val id: Int,
+    val name: String,
+    @SerialName("profile_path")
+    val profilePicture: String
+)
+
+@Serializable
+private class JsonMovie(
+    val id: Int,
+    val title: String,
+    @SerialName("poster_path")
+    val posterPicture: String,
+    @SerialName("backdrop_path")
+    val backdropPicture: String,
+    val runtime: Int,
+    @SerialName("genre_ids")
+    val genreIds: List<Int>,
+    val actors: List<Int>,
+    @SerialName("vote_average")
+    val ratings: Float,
+    @SerialName("vote_count")
+    val votesCount: Int,
+    val overview: String,
+    val adult: Boolean
+)
+
+private suspend fun loadGenres(context: Context): List<Genre> = withContext(Dispatchers.IO) {
+    val data = readAssetFileToString(context, "genres.json")
+    parseGenres(data)
+}
+
+internal fun parseGenres(data: String): List<Genre> {
```

Comment:
@AleksandrAlex Please use coroutines for parsing too. You need to do it on `Dispachers.Default`

Reply:
JsonLoad is not my file! Should I edit it???